src/client/api/events/sub_serialized_output.c

```diff
@@ -4,29 +4,45 @@
 #include <string.h>
 
 #include "client/api/events/sub_serialized_output.h"
+#include "core/address.h"
+#include "core/models/payloads/tagged_data.h"
+#include "core/utils/macros.h"
 
 int event_sub_txn_included_msg(event_client_handle_t client, int *mid, char const transaction_id[], int qos) {
-  if ((strlen(transaction_id)) != EVENT_TXN_ID_LEN) {
+  if ((strlen(transaction_id)) != BIN_TO_HEX_BYTES(IOTA_TRANSACTION_ID_BYTES)) {
     printf("[%s:%d]: Transaction id length is invalid\n", __func__, __LINE__);
     return -1;
   }
   // 95 is the max length for string transactions/{transactionId}/included-message
-  char topic_buff[95] = {0};
+  char topic_buff[97] = {0};
 
-  sprintf(topic_buff, "transactions/%s/included-message", transaction_id);
+  sprintf(topic_buff, "transactions/0x%s/included-message", transaction_id);
 
   return event_subscribe(client, mid, topic_buff, qos);
 }
 
-int event_sub_msg_indexation(event_client_handle_t client, int *mid, char const index[], int qos) {
-  if (((strlen(index)) > EVENT_MS_INDEX_MAX_LEN) || ((strlen(index)) < EVENT_MS_INDEX_MIN_LEN)) {
-    printf("[%s:%d]: Index length is invalid\n", __func__, __LINE__);
+int event_sub_tx_msg_tagged_data(event_client_handle_t client, int *mid, char const tag[], int qos) {
+  if (((strlen(tag)) > TAGGED_DATA_TAG_MAX_LENGTH_BYTES)) {
+    printf("[%s:%d]: Tag length is invalid\n", __func__, __LINE__);
     return -1;
   }
-  // 85 is the max length for string messages/indexation/{index}, index max size is 64 bytes
-  char topic_buff[85] = {0};
+  // 98 is the max length for string messages/transaction/tagged-data/{tag}, tag max len is 64 bytes
+  char topic_buff[98] = {0};
 
-  sprintf(topic_buff, "messages/indexation/%s", index);
+  sprintf(topic_buff, "messages/transaction/tagged-data/%s", tag);
+
+  return event_subscribe(client, mid, topic_buff, qos);
+}
+
+int event_sub_msg_tagged_data(event_client_handle_t client, int *mid, char const tag[], int qos) {
+  if (((strlen(tag)) > TAGGED_DATA_TAG_MAX_LENGTH_BYTES)) {
```

Comment:
Two parentheses too many.
```suggestion
  if (strlen(tag) > TAGGED_DATA_TAG_MAX_LENGTH_BYTES) {
```